dd-sdk-android-gradle-plugin/src/test/kotlin/com/datadog/gradle/plugin/DdAndroidGradlePluginFunctionalTest.kt

```diff
@@ -939,9 +945,127 @@ internal class DdAndroidGradlePluginFunctionalTest {
 
     // endregion
 
+    // region Symbol Upload
+
+    @Test
+    fun `M not try to upload the symbol file W no cmake dependencies { using a fake API_KEY }`(forge: Forge) {
+        // Given
+        stubGradleBuildFromResourceFile(
+            "build_with_datadog_dep.gradle",
+            appBuildGradleFile
+        )
+        stubNativeFiles()
+
+        val result = gradleRunner { withArguments("--info", ":samples:app:assembleRelease") }
+            .build()
+
+        // Then
+        assertThat(result).hasNoNdkSymbolUploadTasks()
+    }
+
+    @Test
+    fun `M try to upload the symbol file W upload { using a fake API_KEY }`(forge: Forge) {
```

Comment:
We need also to add the test where we do JVM mapping upload and NDK symbols upload, verifying that **all** uploads have the same `build_id` and it matches the `build_id` in the APK/AAB. This is very important for the consistency, because all uploads for the same build (variant) should share the same `build_id`.